From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 15 Mar 2016 11:33:52 +0100
Subject: [PATCH] Install vgabios (and link) for aarch64

vgabios is needed for standard VGA mode in AArch64 VMs, so
install it by default on AArch64 and create missing link
in </usr/share/qemu> to </usr/share/vgabios/vgabios.bin>.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Signed-off-by: Stanislaw Kardach <stanislaw.kardach@caviumnetworks.com>
---
 deployment/puppet/openstack_tasks/manifests/roles/compute.pp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/deployment/puppet/openstack_tasks/manifests/roles/compute.pp b/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
index ca1b2c4..c0349dc 100644
--- a/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
+++ b/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
@@ -364,3 +368,11 @@ class openstack::compute (
     libvirt_service_name                       => $::nova::params::libvirt_service_name,
+  } ->
+  package { 'vgabios':
+    ensure => present;
+  } ->
+  file { '/usr/share/qemu/vgabios-stdvga.bin':
+    ensure  => link,
+    target  => '/usr/share/vgabios/vgabios.bin',
+    replace => false,
   }
 
